### PR TITLE
Don't intern all strings and numbers.  Just the ones used as declaration names

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -5775,6 +5750,52 @@ module ts {
...
+                    break;
+                case SyntaxKind.StringLiteral:
+                case SyntaxKind.NumericLiteral:
+                    // We want to store any numbers/strings if they were a name that could be
+                    // related to a declaration.  So, if we have 'import x = require("something")'
+                    // then we want 'something' to be in the name table.  Similarly, if we have
+                    // "a['propname']" then we want to store "propname" in the name table.
+                    if (isDeclarationName(node) ||
+                        node.parent.kind === SyntaxKind.ExternalModuleReference ||
+                        isArgumentOfElementAccessExpression(node)) {
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Wouldn't you be more curious as to whether the element access is part of the LHS of an assignment?

Comment (CyrusNajmabadi) [inline src/services/services.ts]: No.  For example, if you have:

```
foo(a['propname'])
```

we simply want to add 'propname' into the name table any time we see this, so we can effectively do 'Find All Refs' on a property called 'propname'.

Comment (DanielRosenwasser) [inline src/services/services.ts]: Ah, sorry, for some reason I was only thinking of go-to-def.

**Comment by DanielRosenwasser** [comment]
:+1:

**Pull request merged**